### PR TITLE
Support `datetime.timedelta` as a valid wait unit type

### LINKED ISSUE #341: Support timedelta as a valid type for `wait` classes
Currently, the `wait`ing classes only take `int` or `float` as their wait unit types where they are representing seconds.
It would make for a better, more readable code if I could simply pass in a `datetime.timedelta` instead and thus also avoid passing in `wait=60*5` for 5 minutes and instead could `wait=timedelta(minutes=5)`

### PR DESCRIPTION
**Pull request by noambloom**
Closes #341

### CODE REVIEW

**Comment by noambloom** [comment]
@jd do you mind taking a look please? Thanks :)

File: doc/source/index.rst [hunk 1]
```diff
@@ -213,9 +213,9 @@ exceptions, as in the cases here.
         print("Retry forever with no wait if an IOError occurs, raise any other errors")
         raise Exception
 
-    @retry(retry=retry_if_not_exception_type(ClientError))
+    @retry(retry=retry_if_not_exception_type(ConnectionError))
     def might_client_error():
-        print("Retry forever with no wait if any error other than ClientError occurs. Immediately raise ClientError.")
+        print("Retry forever with no wait if any error other than ConnectionError occurs. Immediately raise ConnectionError.")
```
Comment (jd) [inline doc/source/index.rst:218]: This seems unrelated and has been fixed in a different PR.

**Review by jd — DISMISSED** [review]

File: tenacity/wait.py [hunk 2]
```diff
@@ -141,14 +148,14 @@ class wait_exponential(wait_base):
 
     def __init__(
         self,
-        multiplier: typing.Union[int, float] = 1,
-        max: typing.Union[int, float] = _utils.MAX_WAIT,  # noqa
+        multiplier: wait_unit_type = 1,
```
Comment (jd) [inline tenacity/wait.py:151]: This is not time, just a multiplier.

**Review by jd — DISMISSED** [review]
Might be nice to have a couple of tests also, even if typing makes it work in theory.

**Review by jd — DISMISSED** [review]

**Review by jd — APPROVED** [review]
